RealmSwift/RealmCollection.swift

```diff
@@ -138,6 +138,7 @@ public protocol RealmCollectionValue {
 #if swift(>=3.2)
 /// :nodoc:
 public protocol RealmCollectionBase: RandomAccessCollection, LazyCollectionProtocol, CustomStringConvertible, ThreadConfined where Element: RealmCollectionValue {
+    associatedtype ElementType: RealmCollectionValue
```

Comment:
What problem is this fixing? Is there a test that catches it? Can we split it into its own PR since it seems unrelated to this issue?

Reply:
The whole point of RealmCollectionBase was to avoid having this associatedtype, as the compiler warned that it was redundant. If that's no longer the case then RCB should be removed entirely.

Reply:
Yes. I will investigate this further and break out any needed changes into a separate PR.